core/src/main/java/com/ultramixer/jarbundler/PropertyListWriter.java

```diff
@@ -281,90 +270,141 @@ private void buildDOM()
+            {
+                writeClasspath(classPath, extraClassPath, javaDict);
+            }
 
 
-        // JVM options, optional
-        if (bundleProperties.getVMOptions() != null)
-        {
-            writeKeyStringPair("VMOptions", bundleProperties.getVMOptions(), javaDict);
-        }
+            // JVM options, optional
+            if (bundleProperties.getVMOptions() != null)
+            {
+                writeKeyStringPair("VMOptions", bundleProperties.getVMOptions(), javaDict);
+            }
 
-        // Working directory, optional
-        if (bundleProperties.getWorkingDirectory() != null)
-        {
-            writeKeyStringPair("WorkingDirectory", bundleProperties.getWorkingDirectory(), javaDict);
-        }
+            // Working directory, optional
+            if (bundleProperties.getWorkingDirectory() != null)
+            {
+                writeKeyStringPair("WorkingDirectory", bundleProperties.getWorkingDirectory(), javaDict);
+            }
 
-        // StartOnMainThread, optional
-        if (bundleProperties.getStartOnMainThread() != null)
-        {
-            writeKey("StartOnMainThread", javaDict);
-            createNode(bundleProperties.getStartOnMainThread().toString(), javaDict);
-        }
+            // StartOnMainThread, optional
+            if (bundleProperties.getStartOnMainThread() != null)
+            {
+                writeKey("StartOnMainThread", javaDict);
+                createNode(bundleProperties.getStartOnMainThread().toString(), javaDict);
+            }
 
-        // SplashFile, optional
-        if (bundleProperties.getSplashFile() != null)
-        {
-            writeKeyStringPair("SplashFile", bundleProperties.getSplashFile(), javaDict);
-        }
+            // SplashFile, optional
+            if (bundleProperties.getSplashFile() != null)
+            {
+                writeKeyStringPair("SplashFile", bundleProperties.getSplashFile(), javaDict);
+            }
 
-        // Main class arguments, optional
-        if (bundleProperties.getArguments() != null)
-        {
-            writeKeyStringPair("Arguments", bundleProperties.getArguments(), javaDict);
-        }
+            // Main class arguments, optional
+            if (bundleProperties.getArguments() != null)
+            {
+                writeKeyStringPair("Arguments", bundleProperties.getArguments(), javaDict);
+            }
 
-        // Java properties, optional
-        Hashtable javaProperties = bundleProperties.getJavaProperties();
+            // Java properties, optional
+            Hashtable javaProperties = bundleProperties.getJavaProperties();
 
-        if (javaProperties.isEmpty() == false)
+            if (javaProperties.isEmpty() == false)
+            {
+                writeJavaProperties(javaProperties, javaDict);
+            }
+        }
+        else
         {
-            writeJavaProperties(javaProperties, javaDict);
+            // Oracle Java Version
```

Comment:
You're introducing the Oracle Plist style here, depending only on the "required" JVM version without an extra setter.

This will break quite a lot of build pipelines and apps out there which rely on this project and the exact Plist structure.

I don't have objections against the Oracle style in general.  However, if we decide to support this format, there should be an attribut / a setter for this to explicitly turn this feature on and don't break old build scripts.

Reply:
I don't understand. But since I don't understand, I withdraw.